### PR TITLE
event: support `X-Consul-Results-Filtered-By-ACLs` header in list endpoint

### PR DESCRIPTION
**Pull request by boxofrad**
Adds support for the `X-Consul-Results-Filtered-By-ACLs` HTTP header (introduced in #11569) to the event list endpoint.

### CODE REVIEW

File: agent/event_endpoint.go [hunk 1]
```diff
@@ -127,25 +127,44 @@ RUN_QUERY:
...
 
+	// Filter the events if requested
+	if nameFilter != "" {
+		for i := 0; i < len(events); i++ {
+			if events[i].Name != nameFilter {
+				events = append(events[:i], events[i+1:]...)
+				i--
+			}
+		}
+	}
```
Comment (FFMMM) [inline agent/event_endpoint.go:138]: Just curious, why are we doing the name filter before the ACL filter?

Comment (boxofrad) [reply agent/event_endpoint.go:138]: See comment below (on L142) 😄 

```Go
// Note: we filter the results with ACLs *after* applying the user-supplied
// name filter, to ensure the filtered-by-acls header we set below does not
// include results that would be filtered out even if the user did have
// permission.
```

**Review by FFMMM — APPROVED** [review]
🚢

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/515999.

**Pull request merged**